ddtrace/tracer/spancontext.go

```diff
@@ -530,7 +527,17 @@ func (t *trace) finishedOne(s *span) {
 }
 
 func (t *trace) finishChunk(tr *tracer, ch *chunk) {
-	atomic.AddUint32(&tr.spansFinished, uint32(len(ch.spans)))
+	for _, sp := range ch.spans {
+		if sp == nil {
+			continue
+		}
+		tr.spansFinished.mu.Lock()
+		if tr.spansFinished.spans == nil {
+			tr.spansFinished.spans = make(map[string]uint32)
```

Comment:
I believe the `uint32` type is just leftover from when we were using atomics, right? With your current implementation, does this type still need to be uint32? You ultimately convert it to a int64.